cpp/src/replace/replace.cu

```diff
@@ -182,51 +192,107 @@ void replace_nulls_with_column(gdf_size_type size, Type* out_data, gdf_valid_typ
  *        `replace_nulls` with the apropiate data types.
  */
 /* ----------------------------------------------------------------------------*/
-struct replace_nulls_kernel_forwarder {
+struct replace_nulls_column_kernel_forwarder {
   template <typename col_type>
-  void operator()(gdf_size_type           nrows,
-                  gdf_size_type           new_values_length,
-                  void*            d_col_data,
-                  gdf_valid_type*  d_col_valid,
-                  const void*      d_new_value)
+  void operator()(gdf_size_type    nrows,
+                  void*            d_in_data,
+                  gdf_valid_type*  d_in_valid,
+                  const void*      d_replacement,
+                  void*            d_out_data)
   {
     const size_t grid_size = nrows / BLOCK_SIZE + (nrows % BLOCK_SIZE != 0);
-    if (new_values_length == 1) {
-      replace_nulls_with_scalar<<<grid_size, BLOCK_SIZE>>>(nrows,
-                                            static_cast<col_type*>(d_col_data),
-                                            (d_col_valid),
-                                            static_cast<const col_type*>(d_new_value)
-                                            );
-    } else if(new_values_length == nrows) {
-      replace_nulls_with_column<<<grid_size, BLOCK_SIZE>>>(nrows,
-                                            static_cast<col_type*>(d_col_data),
-                                            (d_col_valid),
-                                            static_cast<const col_type*>(d_new_value)
-                                            );
-      
-    }
+
+    replace_nulls_with_column<<<grid_size, BLOCK_SIZE>>>(nrows,
+                                          static_cast<const col_type*>(d_in_data),
+                                          (d_in_valid),
+                                          static_cast<const col_type*>(d_replacement),
+                                          static_cast<col_type*>(d_out_data)
+                                          );
+
+  }
+};
+
+
+/* --------------------------------------------------------------------------*/
+/** 
+ * @brief Functor called by the `type_dispatcher` in order to invoke and instantiate
+ *        `replace_nulls` with the apropiate data types.
+ */
+/* ----------------------------------------------------------------------------*/
+struct replace_nulls_scalar_kernel_forwarder {
+  template <typename col_type>
+  void operator()(gdf_size_type    nrows,
+                  void*            d_in_data,
+                  gdf_valid_type*  d_in_valid,
+                  const void*      replacement,
+                  void*            d_out_data)
+  {
+    const size_t grid_size = nrows / BLOCK_SIZE + (nrows % BLOCK_SIZE != 0);
```

Comment:
Use this instead: https://github.com/rapidsai/cudf/blob/branch-0.8/cpp/src/utilities/cuda_utils.hpp#L39